bottleneck/csrc/move_median.h

```diff
@@ -4,6 +4,16 @@
 #include <math.h>
 #include <assert.h>
 
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
+#define inline __inline
+static __inline float __NAN() {
+  float value;
+  memset(&value, 0xFF, sizeof(value));
+  return value;
+}
+#define NAN __NAN()
```

Comment:
@jennolsen84, I had trouble redefining NAN using `const double NAN = __NAN()`. I ended up trying:
```
kg@kg:/devel/bottleneck (cgohlke-patch-1 *)$ di
diff --git a/bottleneck/csrc/move_median.c b/bottleneck/csrc/move_median.c
index cf94d83..ff5d477 100644
--- a/bottleneck/csrc/move_median.c
+++ b/bottleneck/csrc/move_median.c
@@ -10,6 +10,11 @@ node1->idx = idx2;                                 \
 node2->idx = idx1;                                 \
 idx1       = idx2
 
+/* define nan */
+ai_t myNAN;
+void init_nan(void) {
+  memset(&myNAN, 0xFF, sizeof(ai_t));
+}
 
 /*
 -----------------------------------------------------------------------------
@@ -64,6 +69,8 @@ mm_new(const idx_t window, idx_t min_count)
 
     mm_reset(mm);
 
+    init_nan();
+
     return mm;
 }
 
@@ -511,7 +518,7 @@ mm_get_median(mm_handle *mm)
 {
     idx_t n_total = mm->n_l + mm->n_s;
     if (n_total < mm->min_count)
-        return NAN;
+        return myNAN;
     if (min(mm->window, n_total) % 2 == 1)
         return mm->s_heap[0]->ai;
     return (mm->s_heap[0]->ai + mm->l_heap[0]->ai) / 2.0;
diff --git a/bottleneck/csrc/move_median.h b/bottleneck/csrc/move_median.h
index d6bbe54..736b0b0 100644
--- a/bottleneck/csrc/move_median.h
+++ b/bottleneck/csrc/move_median.h
@@ -6,12 +6,6 @@
 
 #if defined(_MSC_VER) && (_MSC_VER < 1900)
 #define inline __inline
-static __inline float __NAN() {
-  float value;
-  memset(&value, 0xFF, sizeof(value));
-  return value;
-}
-#define NAN __NAN()
 #endif
 
 #if NOCYTHON==1
```

Note that NAN is only used in one place---calculating the median when the count is less than `min_count`.  These changes slow the code a bit, but I wasn't too careful in my measurement.